File: angular/src/app/model/adresse.ts
export class Adresse {
  constructor(private _numero?: number, private _rue?: string, private _cp?: number, private _ville?: string) {

  }

  get numero(): number {
    return this._numero;
  }

  set numero(value: number) {
    this._numero = value;
  }

  get rue(): string {
    return this._rue;
  }

  set rue(value: string) {
    this._rue = value;
  }

  get cp(): number {
    return this._cp;
  }

  set cp(value: number) {
    this._cp = value;
  }

  get ville(): string {
    return this._ville;
  }

  set ville(value: string) {
    this._ville = value;
  }
}

File: java/src/main/java/com/sopra/Factory/model/Adresse.java
package com.sopra.Factory.model;

import javax.persistence.Column;
import javax.persistence.Embeddable;

import com.fasterxml.jackson.annotation.JsonView;

@Embeddable
public class Adresse {

	@JsonView(JsonViews.Common.class)
	@Column(name = "numero")
	private Integer numero;

	@Column(name = "rue")
	@JsonView(JsonViews.Common.class)
	private String rue;

	@Column(name = "code_postal")
	@JsonView(JsonViews.Common.class)
	private Integer cp;

	@Column(name = "ville")
	@JsonView(JsonViews.Common.class)
	private String ville;

	public Adresse() {
		super();
	}

	public Integer getNumero() {
		return numero;
	}

	public void setNumero(Integer numero) {
		this.numero = numero;
	}

	public String getRue() {
		return rue;
	}

	public void setRue(String rue) {
		this.rue = rue;
	}

	public Integer getCp() {
		return cp;
	}

	public void setCp(Integer cp) {
		this.cp = cp;
	}

	public String getVille() {
		return ville;
	}

	public void setVille(String ville) {
		this.ville = ville;
	}

}
